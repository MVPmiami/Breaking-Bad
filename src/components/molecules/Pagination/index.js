import styles from "./style.module.scss";
import PageNumberBtn from "../../atoms/PageNumberBtn";
import ArrowNavigation from "../../atoms/ArrowNavigation";
import ShowCards from "../../atoms/ShowCards";

const Pagination = ({
  personPerPage,
  totalPersons,
  paginate,
  currentPage,
  paginateToNextPage,
  paginateToPrevPage,
  changeAmountCards,
}) => {
  let pageNumbers = [];
  for (let i = 1; i <= Math.ceil(totalPersons / personPerPage); i++) {
    pageNumbers.push(i);
  }
  const dots = "...";

  if (pageNumbers.length > 6) {
    let lastPageNumber = pageNumbers[pageNumbers.length - 1];
    pageNumbers = [
      currentPage,
      currentPage + 1,
      currentPage + 2,
      dots,
      lastPageNumber,
    ];
    if (currentPage >= 10) {
      pageNumbers = [10, 11, lastPageNumber];
    }
  } else if (pageNumbers.length > 5 && pageNumbers.length <= 6) {
    let lastPageNumber = pageNumbers[pageNumbers.length - 1];
    pageNumbers = [
      currentPage,
      currentPage + 1,
      currentPage + 2,
      dots,
      lastPageNumber,
    ];
    if (currentPage >= 4) {
      pageNumbers = [4, 5, lastPageNumber];
    }
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.leftSide}>
        {pageNumbers.map((numberPage) => (
          <PageNumberBtn
            currentPage={currentPage}
            numberPage={numberPage}
            paginate={paginate}
          />
        ))}
        <ArrowNavigation
          currentPage={currentPage}
          paginateToPrevPage={paginateToPrevPage}
          paginateToNextPage={paginateToNextPage}
        />
      </div>
      <div className={styles.rightSide}>
        <ShowCards
          personPerPage={personPerPage}
          changeAmountCards={changeAmountCards}
        />
      </div>
    </div>
  );
};

export default Pagination;
