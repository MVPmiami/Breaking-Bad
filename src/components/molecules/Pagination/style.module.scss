@import "./../../../assets/styles/style.scss";

.wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10rem;
  height: 4rem;
  width: 100%;
  .leftSide {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 4rem;
  }
  .rightSide {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 16rem;
    height: 4rem;
  }
}

@media (max-width: 1215px) {
  .wrapper {
    flex-flow: column;
    height: 8rem;
    align-items: flex-start;
  }
}
