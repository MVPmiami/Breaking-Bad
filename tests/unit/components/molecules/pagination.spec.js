import React from "react";
import Pagination from "../../../../src/components/molecules/Pagination";

const setUp = (props) => shallow(<Pagination {...props} />);

describe("should render Pagination component", () => {
  let component;
  let totalPersons = 50;
  let personPerPage = 10;
  let currentPage = 1;
  let paginateToNextPage = jest.fn();
  let paginateToPrevPage = jest.fn();
  let changeAmountCards = jest.fn();
  beforeEach(() => {
    component = setUp({
      totalPersons,
      personPerPage,
      currentPage,
      paginateToNextPage,
      paginateToPrevPage,
      changeAmountCards,
    });
  });

  it("should contain .cardWrapper wrapper", () => {
    expect(component).toMatchSnapshot();
  });
});
