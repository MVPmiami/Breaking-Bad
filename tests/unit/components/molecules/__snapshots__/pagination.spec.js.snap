// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination component should contain .cardWrapper wrapper 1`] = `
<div
  className="wrapper"
>
  <div
    className="leftSide"
  >
    <PageNumberBtn
      currentPage={1}
      key="1"
      numberPage={1}
    />
    <PageNumberBtn
      currentPage={1}
      key="2"
      numberPage={2}
    />
    <PageNumberBtn
      currentPage={1}
      key="3"
      numberPage={3}
    />
    <PageNumberBtn
      currentPage={1}
      key="4"
      numberPage={4}
    />
    <PageNumberBtn
      currentPage={1}
      key="5"
      numberPage={5}
    />
    <ArrowNavigation
      currentPage={1}
      paginateToNextPage={[MockFunction]}
      paginateToPrevPage={[MockFunction]}
    />
  </div>
  <div
    className="rightSide"
  >
    <ShowCards
      changeAmountCards={[MockFunction]}
      personPerPage={10}
    />
  </div>
</div>
`;
